types/eve.go

```diff
@@ -270,3 +272,46 @@ type EveEvent struct {
 	Stats            *statsEvent    `json:"stats,omitempty"`
 	ExtraInfo        *ExtraInfo     `json:"_extra,omitempty"`
 }
+
+// EveOutEvent is the version of EveEvent that we use to marshal the output for
+// downstream consumption.
+type EveOutEvent EveEvent
+
+// MarshalJSON for EveOutEvents ensures that FlowIDs are represented in JSON
+// as a string. This is necessary to work around some arbitrary limitations such
+// as syslog-ng's funny JSON parser implementation, which truncates large
+// integers found in JSON values.
+func (e EveOutEvent) MarshalJSON() ([]byte, error) {
+	type Alias EveOutEvent
+	v, err := json.Marshal(&struct {
+		FlowID string `json:"flow_id"`
+		Alias
+	}{
+		FlowID: fmt.Sprintf("%d", e.FlowID),
```

Comment:
`fmt` is okish, but  `strconv.FormatInt(i, 10)` is better.

Reply:
True.